### ISSUE TITLE
How to pass object into vm, execute object functions and return it back?

### ISSUE THREAD

**Issue by 1bye** [laverdet/isolated-vm#358]
Hello! I'm glad using this awesome library. I'm currently trying to migrate from vm2 to isolated-vm, and I'm having trouble understanding how to pass an object to the VM. Here's an example object that I want to work with:
```js
{
  status: 'Not Worked',
  changeStatus(status) {
     this.status = status;
  }
}
```
I would like to execute some code like `( () => { app.changeStatus('Worked') } )();` and retrieve the object with the changed value `{ status: 'Worked' }`. Could you please help me understand how to achieve this using isolated-vm?

I've tried this, but unfortunately this don't work:
```js
const isolate = new ivm.Isolate({ memoryLimit: 8 /* MB */ });

const _code = `(() => { app.changeStatus('Worked') })()`;
const script = await isolate.compileScript(_code)
const context = await isolate.createContext();

const app = { status: 'Not worked', changeStatus(status) {this.status = status} };

await context.global.set('app', app);

return script.runSync(context);
```

**Thanks!**

**Comment by mklinke** [comment]
I've just started looking into the migration from vm2 to isolated-vm, too. Take a look into [this comment](https://github.com/laverdet/isolated-vm/issues/322#issuecomment-1339071261) for some information that helped me proceed with my POC. 

TL;DR: The integration between isolates is (a little) more effort than with vm2 and you have to think about references, copies and which code executes where. If you don't need side effects (I/O), you may get away with adding the code to the global context of the new isolate itself. However, if you want to allow the isolated code to use side effects, you can create a reference to a function that you have to call in a specific way. See the linked comment for more details.

HTH and good luck!

**Comment by JonasJonny** [comment]
Hi @1bye, did you found the solution and is it possible to share the result?
Thank you in advance.
